fenwick/src/main/java/org/opencadc/fenwick/StorageSiteSync.java

```diff
@@ -0,0 +1,204 @@
+            if (storageSiteIterator.hasNext()) {
+                final StorageSite storageSite = storageSiteIterator.next();
+                if (storageSiteIterator.hasNext()) {
+                    throw new IllegalStateException("More than one Storage Site found.");
+                } else {
+                    putStorageSite(storageSite);
+                    return storageSite;
+                }
+            } else {
+                throw new IllegalStateException("No storage sites available to sync.");
+            }
+        }
+    }
+
+    /**
+     * Store the StorageSite in the local inventory database.
+     * @param storageSite   The StorageSite to store.
+     */
+    void putStorageSite(final StorageSite storageSite) {
+        final TransactionManager transactionManager = storageSiteDAO.getTransactionManager();
+
+        try {
+            log.debug("Start transaction.");
+            transactionManager.startTransaction();
+            storageSiteDAO.put(storageSite);
+            transactionManager.commitTransaction();
+            log.debug("Commit transaction.");
+            transactionManager.commitTransaction();
+        } catch (Exception e) {
+            log.error(String.format("Failed to put StorageSite %s.", storageSite), e);
+            transactionManager.rollbackTransaction();
+            log.debug("Rollback Transaction: OK");
+            throw e;
+        } finally {
+            if (transactionManager.isOpen()) {
+                log.error("BUG - Open transaction in finally");
+                transactionManager.rollbackTransaction();
+                log.error("Transaction rolled back successfully.");
+            }
+        }
+    }
+
+    /**
+     * Override this method in tests to avoid recreating a TapClient.
+     * @return      Iterator over Storage Sites, or empty Iterator.  Never null.
+     * @throws AccessControlException permission denied
+     * @throws NotAuthenticatedException authentication attempt failed or rejected
+     * @throws ByteLimitExceededException input or output limit exceeded
+     * @throws IllegalArgumentException null method arguments or invalid query
+     * @throws ResourceNotFoundException remote resource not found
+     * @throws TransientException temporary failure of TAP service: same call could work in future
+     * @throws IOException failure to send or read data stream
+     * @throws InterruptedException thread interrupted
+     */
+    ResourceIterator<StorageSite> queryStorageSites() throws AccessControlException, ResourceNotFoundException,
+                                                             ByteLimitExceededException, NotAuthenticatedException,
+                                                             IllegalArgumentException, TransientException, IOException,
+                                                             InterruptedException {
+        return tapClient.execute(STORAGE_SITE_QUERY, row -> {
```

Comment:
This also needs to assign the entity fields (lastModified and metaChecksum) using InventoryUtil.asssignLastModified and InventoryUtil.assignMetaChecksum respectively.